pkg/api/defaults.go

```diff
@@ -46,6 +46,15 @@ func SetDefaults(doc *OpenShiftClusterDocument) {
 		if doc.OpenShiftCluster.Properties.NetworkProfile.PreconfiguredNSG == "" {
 			doc.OpenShiftCluster.Properties.NetworkProfile.PreconfiguredNSG = PreconfiguredNSGDisabled
 		}
+
+		// If OutboundType is Loadbalancer and there is no LoadBalancerProfile, set default one
```

Comment:
I'm not entirely clear on why `ManagedOutboundIP` needs to be set if the setup is default (OutboundType is Loadbalancer and there is no LoadBalancerProfile). Am I missing something?

Reply:
Mainly to be explicit on the configuration of the public load balancer.  If this is not set here then it will be nil and not be represented properly to the customer on the return payload of  a put/patch operation.  I think it would be confusing if the field `effectiveOutboundIPs` will be present and showing the 1 outbound IP on the load balancer but none of the outbound IP options are represented. 

If I don't set it as a default the cluster doc looks like this:
```
"loadBalancerProfile" : {
   "effectiveOutboundIps" : [
      {"resourceIdOfDefaultPublicIP"}
   ]
}
```
If set the cluster doc looks like this:
```
"loadBalancerProfile" : {
   "effectiveOutboundIps" : [
      {"resourceIdOfDefaultPublicIP"}
   ]
   "managedOutboundIps": {
      "count": 1
   }
}
```

I also added some static validation to make sure the customer can't null all the fields, which I think will be bugged if `effectiveOutboundIps` is set but the others aren't.
```
"loadBalancerProfile" : {
   "effectiveOutboundIps" : [
      {"resourceIdOfDefaultPublicIP"}
   ], 
   # Invalid, at least one outboundIP type must be active at all times
   "managedOutboundIps": null,
   "outboundIps" : null,
   "outboundIpPrefixes": null
}
```

Reply:
> I think it would be confusing if the field effectiveOutboundIPs will be present and showing the 1 outbound IP on the load balancer but none of the outbound IP options are represented.

If it would be confusing to set some fields of loadBalancerProfile and not others, wouldn't we want to keep loadBalancerProfile nil If OutboundType is Loadbalancer? To me one of these would make the most sense:

```
"loadBalancerProfile" : {} // is outboundType LoadBalancer
```
OR
```
"loadBalancerProfile" : {
   "effectiveOutboundIps" : [
      {"resourceIdOfDefaultPublicIP"}
   ]
   "managedOutboundIps": {
      "count": 1
   }
} // is outboundType LoadBalancer
```
Whereas now, we have:
```
"loadBalancerProfile" : {
   "managedOutboundIps": {
      "count": 1
   }
} // is outboundType LoadBalancer
```

Reply:
Yes, the second one is the ideal state.  The issue is that `effectiveOutboundIps` is unkown at this point of the code.  My plan to add a step to the install process / update process to update `effectiveOutboundIps`.  
Also I have also been using the aks user experience as a guideline here and this is exactly how aks behaves during.

Reply:
Ok - so we are going to populate it, but can't yet. Makes sense to me 👍